manifests/params.pp

```diff
@@ -24,7 +24,7 @@
   }
 
   # Amazon Linux already have aws cli installed:
-  if getvar('::ec2_metadata') and $::operatingsystem != 'Amazon' {
+  if getvar('::ec2_metadata') != '' and $::operatingsystem != 'Amazon' {
```

Comment:
Tagging @nanliu. Is using getvar a way to also get around strict_variables? It says it's the same as $::, so it seems like a cop-out.